src/main.ts

```diff
@@ -285,17 +289,30 @@ function getCheckoutInfo(ref: string, commit: string): ICheckoutInfo {
   }
   // refs/pull/
   else if (upperRef.startsWith('REFS/PULL/')) {
-    const prNumber = ref.split('/')[2]
-    if (prNumber) {
-      result.ref = `refs/pull/${prNumber}/head`
-    } else {
-      result.ref = ref
-    }
+    const branch = ref.substring('refs/pull/'.length)
+    result.ref = `refs/remotes/pull/${branch}`
   }
   // refs/tags/
   else if (upperRef.startsWith('REFS/')) {
     result.ref = ref
   }
+  // refs/
+  else if (upperRef.startsWith('REFS/')) {
```

Comment:
looks impossible to ne omtered necaise we're already checking for the same condition in L296